### ISSUE TITLE
Output of train score

### ISSUE THREAD

**Issue by Chopin2** [pycaret/pycaret#1176]
Using PyCaret, validation score and prediction score are easy to obtain, but how can I obtain train score?
Although it can be seen on learning curve, I want to output train score as a concrete number.

**Comment by Yard1** [comment]
`predict_model()` without `data` argument specified will predict on train (aka holdout) set.

**Comment by Chopin2** [comment]
Sorry for my bad writing.
I wrote "prediction score" in the sense of "hold out" (or "test") score.
I want to know how to output the score of train set which was used to train the model, like the one which I can see on learning curve (written as "Training score").

**Comment by Yard1** [comment]
I see - there isn't a simple way to do it right now, but it wouldn't be hard to add. Tagging as enhancement.

**Comment by pycaret** [comment]
@Chopin2 Thanks for your input. What's the use case to know the performance metrics on the train set? Just trying to understand...

**Comment by Chopin2** [comment]
I want to know whether overfitting is occurring or not quickly after executing 'compare_models' without creating learning curve  for each model independently. Thanks!

**Comment by gm039** [comment]
Hi,

Can I use the below code to get the prediction probability on train data? WIll the probability for train samples be accurate? 

% Importing dataset
from pycaret.datasets import get_data
diabetes = get_data('diabetes')
%Importing module and initializing setup
from pycaret.classification import *
clf1 = setup(data = diabetes, target = 'Class variable')
%creating a model
lr = create_model('lr')
% Predict on holdout set
Prediction_Score_Holdout= predict_model(lr) 
Prediction_Score_Holdout.head()

% Now predicting on all data (including train also)
Prediction_Score_All = predict_model(lr,data=diabetes) 
Prediction_Score_All .head()

**Comment by Yard1** [comment]
The idea is correct, but this will predict on all data. If you want to predict just on train data, I would first split the data into train and test sets before passing them to setup (you can pass the test dataframe to be used instead of splitting the data it has received as an argument there) and then predict on train.

**Comment by SandyBy** [comment]
hi, when I use plot_model(estimator = tuned_catboost, plot = 'auc',scale=3,use_train_data=False) , I'm using the test set to plot roc,right? However, the auc I get here is quite different from what I get from using predict_model(tuned_catboost) and I try to get 95%CI of auc, any ideas what's wrong here, thanks a lot!

**Comment by SandyBy** [comment]
hi, when I use plot_model(estimator = tuned_catboost, plot = 'auc',scale=3,use_train_data=False) , I'm using the test set to plot roc,right? However, the auc I get here is quite different from what I get from using predict_model(tuned_catboost) and I try to get 95%CI of auc, any ideas what's wrong here, thanks a lot!
![屏幕截图 2021-08-24 092307](https://user-images.githubusercontent.com/46986985/130540774-64fde047-c7a6-4908-8ba1-543e15c6b24d.png)
![屏幕截图 2021-08-24 092549](https://user-images.githubusercontent.com/46986985/130540789-ec4d36b0-313e-47ca-991c-8306833994a9.png)

**Comment by SandyBy** [comment]
sorry, it seems problem solved when I try this. 
![屏幕截图 2021-08-24 134119](https://user-images.githubusercontent.com/46986985/130562287-c5b5dea3-4de1-4d00-b448-6da8ab3d045f.png)
![屏幕截图 2021-08-24 134204](https://user-images.githubusercontent.com/46986985/130562309-c1f9ef5c-b401-4357-bc24-ede76fcf8c82.png)

**Comment by Luizgs7** [comment]
Hi, 
@gm039 I got the splited train and test set to use in the predict_model as you suggested:

`
% Importing dataset
from pycaret.datasets import get_data
diabetes = get_data('diabetes')

%Importing module and initializing setup
from pycaret.classification import *
clf1 = setup(data = diabetes, target = 'Class variable')

%creating a model
lr = create_model('lr')

%Get transformed train and test data
X_training_data = get_config(variable="X_train")
X_test_data = get_config(variable="X_test")

%get not tranformed data
data_before_preprocess = get_config(variable="data_before_preprocess")

%get train and test set before tranformation
train_before_preprocess = data_before_preprocess[data_before_preprocess.index.isin(X_training_data.index)]
test_before_preprocess = data_before_preprocess[data_before_preprocess.index.isin(X_test_data.index)]

% Predict on train test
Prediction_Score_train = predict_model(lr,data=train_before_preprocess)
Prediction_Score_train.head()

% Predict on test set
Prediction_Score_test = predict_model(lr,data=test_before_preprocess)
Prediction_Score_test.head()
`

I hope it helps somebody!

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
